clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -636,6 +636,8 @@ private ListOffsetResult fetchOffsetsByTimes(Map<TopicPartition, Long> timestamp
                     List<ConsumerRecord<K, V>> records = fetchRecords(nextInLineFetch, recordsRemaining);
 
                     if (!records.isEmpty()) {
+                        records = records.stream().filter(r -> !(r instanceof NoConsumerRecord)).collect(Collectors.toList());
```

Comment:
Even if `records` might not be empty, we need to filter out the dummy records we added to indicate tx-markers